src/seismometer/data/loader/pipeline.py

```diff
@@ -0,0 +1,158 @@
+        config : ConfigProvider
+            The loaded configuration object.
+        prediction_fn : ConfigOnlyHook
+            A callable taking a ConfigProvider and returning a dataframe.
+            Used to load a (predictions) dataframe based on configuration.
+        event_fn : ConfigOnlyHook, optional
+            A callable taking a ConfigProvider and returning a dataframe.
+            Used to load a (events) dataframe based on configuration.
+        post_predict_fn : ConfigFrameHook, optional
+            A callable taking a ConfigProvider and a (predictions) dataframe and returning a dataframe.
+            Used to do minor transforms of predictions such as type casting.
+        post_event_fn : ConfigFrameHook, optional
+            A callable taking a ConfigProvider and a (events) dataframe and returning a dataframe.
+            Used to do minor transforms of events such as type casting.
+        merge_fn : MergeHook, optional
+            A callable taking a ConfigProvider, a (events) dataframe, and a (predictions) dataframe
+            and returning a dataframe.
+            Used to merge events onto predictions based on configuration.
+        """
+        self.config = config
+
+        self.prediction_fn = prediction_fn
+        self.post_predict_fn = post_predict_fn or _passthru_framehook
+
+        self.event_fn = event_fn
+        self.post_event_fn = post_event_fn or _passthru_framehook
+        self.merge_fn = merge_fn or _passthru_mergehook
+
+        self.prediction_from_memory: ConfigFrameHook = _passthru_framehook
+        self.event_from_memory: ConfigFrameHook = _passthru_framehook
+
+    def load_data(self, prediction_obj: pd.DataFrame = None, event_obj: pd.DataFrame = None) -> pd.DataFrame:
+        """
+        Entry point for loading data for a Seismogram session.
+
+        Can optionally accept a prediction and event frame that are prioritized over loading from configuration.
+
+        Parameters
+        ----------
+        prediction_obj : pd.DataFrame, optional
+            an optional pre-loaded predictions dataframe, if None then will load from configuration, by default None.
+        event_obj : pd.DataFrame, optional
+            an optional pre-loaded events dataframe, if None then will load from configuration, by default None.
+
+        Returns
+        -------
+        pd.DataFrame
+            the loaded and merged dataframe for a Seismogram session.
+        """
+
+        logger.info(f"Importing files from {self.config.config_dir}")
+
+        dataframe = self._load_predictions(prediction_obj)
+        dataframe = self.post_predict_fn(self.config, dataframe)
+        dataframe = self._add_events(dataframe, event_obj)
+
+        return dataframe
+
+    def _load_predictions(self, prediction_obj: pd.DataFrame = None):
+        """Load predictions from config or memory."""
+        if (prediction_obj is None) and (self.prediction_fn is None):
+            return pd.DataFrame()
```

Comment:
Good call; added as a runtime error.  I'd expect this to primarily be the routers fault (no prediction loader) but when intended it will direct to specifying a frame